src/Adapter/Address/CommandHandler/AddManufacturerAddressHandler.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Address\CommandHandler;
+
+use Address;
+use PrestaShop\PrestaShop\Core\Domain\Address\Command\AddManufacturerAddressCommand;
+use PrestaShop\PrestaShop\Core\Domain\Address\CommandHandler\AddManufacturerAddressHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressException;
+use PrestaShop\PrestaShop\Core\Domain\Address\ValueObject\AddressId;
+
+/**
+ * Adds manufacturer address using legacy object model
+ */
+final class AddManufacturerAddressHandler implements AddManufacturerAddressHandlerInterface
+{
+    /**
+     * @param AddManufacturerAddressCommand $command
+     *
+     * @return AddressId
+     *
+     * @throws AddressException
+     */
+    public function handle(AddManufacturerAddressCommand $command)
+    {
+        $address = $this->createAddressFromCommand($command);
+
+        if (false === $address->add()) {
+            throw new AddressException(
+                sprintf('Failed to add new address "%s"', $command->getAddress())
+            );
+        }
+
+        return new AddressId((int) $address->id);
+    }
+
+    /**
+     * @param AddManufacturerAddressCommand $command
+     *
+     * @return Address
+     */
+    private function createAddressFromCommand(AddManufacturerAddressCommand $command)
+    {
+        $address = new Address();
+        if (null !== $command->getManufacturerId()) {
+            $address->id_manufacturer = $command->getManufacturerId()->getValue();
```

Comment:
@matks 
Manufacturer null value should be accepted as a valid choice (as colinegin mentioned in #9337).
When choice is not changed in edit form, this value will be null, as well as if choice where changed to "no manufacturer" (null too).
I.e. Address has manufacturer and user wants to remove it - user selects "no manufacturers". $manufacturerId will get null value and this handler would not update that field.
What would be the best way handling this?